src/support.cpp

```diff
@@ -406,44 +413,18 @@ void AreaSupport::combineSupportInfillLayers(SliceDataStorage& storage)
     }
 }
 
-
-void AreaSupport::generateOutlineInsets(std::vector<Polygons>& insets, Polygons& outline, const unsigned int inset_count, const coord_t wall_line_width_x)
-{
-    for (unsigned int inset_idx = 0; inset_idx < inset_count; inset_idx++)
-    {
-        insets.push_back(Polygons());
-        if (inset_idx == 0)
-        {
-            insets[0] = outline.offset(-wall_line_width_x / 2);
-        }
-        else
-        {
-            insets[inset_idx] = insets[inset_idx - 1].offset(-wall_line_width_x);
-        }
-
-        // optimize polygons: remove unnecessary verts
-        insets[inset_idx].simplify();
-        if (insets[inset_idx].size() < 1)
-        {
-            insets.pop_back();
-            break;
-        }
-    }
-}
-
 void AreaSupport::cleanup(SliceDataStorage& storage)
 {
-    const coord_t support_line_width = Application::getInstance().current_slice->scene.current_mesh_group->settings.get<coord_t>("support_line_width");
-    for (unsigned int layer_nr = 0; layer_nr < storage.support.supportLayers.size(); layer_nr++)
+    const auto support_line_width = Application::getInstance().current_slice->scene.current_mesh_group->settings.get<coord_t>("support_line_width");
+    for (auto& layer : storage.support.supportLayers)
```

Comment:
I dislike these uses of `auto` too. It's a bit of a matter of opinion, but it's useful to know that these are physical sizes (`coord_t`) and layer indexes (`LayerIndex`). And it's also useful for the compiler to break on it if the type happens to change, for instance the type of `supportLayers`.

Reply:
We differ on style ;-) and that is okay it would be a boring world otherwise :-) 
Since my fondness of `auto` deviates from most of the code currently used I will adapt. Maybe only use it for lambda's and the like from now on.

Reply:
The code base uses it for iterators and lambdas sometimes, to prevent the iterator's type repeating the container's type and repeating the types of the parameters in the lambda's type. But for the rest the code base is statically typed.

Reply:
This set of my nerd alert: `auto` is _not_ dynamic typing! It still gets resolved at compile time, and is therefore still static :-p